homeassistant/components/sonos/speaker.py

```diff
@@ -332,6 +332,17 @@ async def _subscribe(
         subscription.auto_renew_fail = self.async_renew_failed
         self._subscriptions.append(subscription)
 
+    async def async_unsubscribe(self) -> None:
+        """Cancel all subscriptions."""
+        for subscription in self._subscriptions:
```

Comment:
Does it make sense to use asyncio.gather with `return_exceptions=True`?

Reply:
Yup.